zerver/lib/markdown/__init__.py

```diff
@@ -1367,22 +1370,27 @@ def handleMatch(self, match: Match[str]) -> Optional[Element]:
         name = orig_syntax[1:-1]
 
         active_realm_emoji: Dict[str, Dict[str, str]] = {}
+        content: Optional[str] = None
         db_data = self.md.zulip_db_data
         if db_data is not None:
             active_realm_emoji = db_data['active_realm_emoji']
+            content = db_data['content']
 
         if self.md.zulip_message and name in active_realm_emoji:
             return make_realm_emoji(active_realm_emoji[name]['source_url'], orig_syntax)
         elif name == 'zulip':
             return make_realm_emoji('/static/generated/emoji/images/emoji/unicode/zulip.png', orig_syntax)
         elif name in name_to_codepoint:
-            return make_emoji(name_to_codepoint[name], orig_syntax)
+            return make_emoji(name_to_codepoint[name], orig_syntax, content=content)
         else:
             return orig_syntax
 
 def content_has_emoji_syntax(content: str) -> bool:
     return re.search(EMOJI_REGEX, content) is not None
 
+def content_has_only_emoji_syntax(content: str) -> bool:
+    return all(map(lambda msg: re.match(EMOJI_REGEX, msg) is not None, content.split(' ')))
```

Comment:
This incorrectly matches messages where every word *starts* with an emoji. It’s also less efficient than it could be: a single regex like `re.fullmatch(fr"\s*(?:{EMOJI_REGEX}\s*)+", content)` should suffice. Neither version handles emoji that were input as Unicode, though.

Reply:
Thanks, the regex seems to work fine for many cases, However fails to validate if a content like `**:smile:**` is passes.(Inside bold/strikethrough markdown).